### PR TITLE
Validate CSV values and their formats

### PR DESCRIPTION
**Pull request by tie-rack**
Pivotal stories: [90227234](https://www.pivotaltracker.com/story/show/90227234), [90227616](https://www.pivotaltracker.com/story/show/90227616)

Validations for each row of CSV imports that cover required columns having values and that each value is of the correct format (e.g, starting with "http://" or is "yes"/"no" or is formatted like "(XXX) XXX-XXXX").

Includes updates to "full-good-run" example files that were less than "fully good".

The way CSVs are specified has been changed wholesale. Instead of a bunch of function calls to set up each importer, there is now a list of maps that describe each CSV and its columns. That list is read and turned in to the importers and validators that get run. These descriptions should be very readable and easily editable by people less familiar with Clojure code, but familiar with the CSVs themselves.

### CODE REVIEW

File: src/vip/data_processor/validation/csv.clj [hunk 1]
```diff
@@ -3,47 +3,294 @@
...
 
 (defn file-name [file]
   (.getName file))
 
 (defn good-filename? [file]
-  (->> file
-       file-name
-       (contains? csv-filenames)))
+  (let [filename (file-name file)]
+    (contains? csv-filenames filename)))
```
Comment (7maples) [inline src/vip/data_processor/validation/csv.clj]: I really like the explicit `contains?`, it's more readable this way

File: src/vip/data_processor/validation/csv.clj [hunk 2]
```diff
@@ -3,47 +3,294 @@
...
-    "precinct_split_electoral_district.txt"
-    "precinct_split_polling_location.txt"
-    "referendum.txt"
-    "referendum_ballot_response.txt"
-    "source.txt"
-    "state.txt"
-    "state_early_vote_site.txt"
-    "street_segment.txt"})
+(defn boolean-value [x]
+  (if (re-find #"\A(?i:yes)\z" x) 1 0))
```
Comment (7maples) [inline src/vip/data_processor/validation/csv.clj]: Do we want to make this more flexible so it can handle whitespace?

Comment (tie-rack) [inline src/vip/data_processor/validation/csv.clj]: Well, the `yes-no` format validator is more strict than even this. The validations spec is pretty unforgiving. If there's whitespace, you're already going to get an error about the value.

File: test/vip/data_processor/validation/csv_test.clj [hunk 3]
```diff
@@ -88,25 +34,80 @@
...
+          (let [result-ctx (format-rule ctx {} line-number)]
+            (is (= ctx result-ctx))))
+        (testing "if it is there"
+          (testing "it matches the format, everything's okay"
+            (let [result-ctx (format-rule ctx {column "1234"} line-number)]
+              (is (= ctx result-ctx))))
+          (testing "it doesn't match the format, you get an error"
+            (let [result-ctx (format-rule ctx {column "asdf"} line-number)]
+              (is (get-in result-ctx [:errors filename line-number column])))))))
+    (testing "check that is a list of options"
```
Comment (7maples) [inline test/vip/data_processor/validation/csv_test.clj]: at first read this was awkward - maybe "list of options check" or "a check that is a list of options"?

File: src/vip/data_processor/validation/csv.clj (hunk 2 above)
Comment (7maples) [inline src/vip/data_processor/validation/csv.clj]: great!

**Comment by 7maples** [comment]
LGTM

**Pull request merged**